### PR TITLE
Deploy the app using a custom docker image

### PR DESCRIPTION
**Pull request by khcr**
I created a custom docker image based on the Jelastic one. The image has been successfully tested in Jelastic cloud.

I suggest we host the image on Github packages. Could you create the action to build the image on master ?
A few remarks:
- the build needs a secret, see .deploy/README.md
- that would be great to incrementaly tag the image because it would let us rollback easily (the default tag is `latest`).
- the Dockerfile contains a `git checkout ci/custom-docker` that can be removed once the PR is merged.

### CODE REVIEW

File: .deploy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+RUN curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
+RUN yum -y install yarn
+
+# Set the context
+ENV RAILS_ENV=${context}
+RUN echo "rails_env ${context};" > /etc/nginx/ruby.env
+
+# Clone the application
+WORKDIR ${WEBROOT}
+RUN rm -rf ROOT
```
Comment (nkcr) [inline .deploy/Dockerfile:23]: I'm not sure where this root comes from..?

Comment (khcr) [reply .deploy/Dockerfile:23]: It comes from the Jelastic image, which already contains a default rails app in $WEBROOT/ROOT.

File: .deploy/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+if [[ -z "${RAILS_MASTER_KEY}" ]]; then
```
Comment (nkcr) [inline .deploy/entrypoint.sh:6]: I'm under the impression that you check is false.
`if [[ -z "${RAILS_MASTER_KEY}" ]]; then`

returns true is the variable is empty.

**Review by nkcr — COMMENTED** [review]
Just a comment over your script. I suggest we merge this PR and the one that introduces the Github action before I create a new one to publish docker images.

File: .deploy/entrypoint.sh (hunk 2 above)
Comment (khcr) [reply .deploy/entrypoint.sh:6]: Indeed, your impression is right.

Comment (nkcr) [reply .deploy/entrypoint.sh:6]: A detail, but you could 've added `exit 1` if the argument is empty, and therefore avoid the `else`, which removes a level of indentation for the rest of the code.

**Pull request merged**